gitbook-docs/rest_api.md

```diff
@@ -1,22 +1,62 @@
-#### REST/HTTP API: /signalk/v1/api/
+# REST API
 
-Note the trailing slash in the path.
+Signal K producers MAY implement an HTTP API which consumers can use to self-configure, poll for Signal K data or make
+configuration changes. As specified in the previous section, all URLs for interacting with Signal K are rooted at
+`/signalk`.
 
-The base URL MUST provide a Signal K document that is valid according to the full Signal K [schema
-specification]({{site.baseurl}}specification.html). The contents SHOULD be all the current values of the data items the
-server knows.
+## GET /signalk
 
-If the path following the base is a valid Signal K path `GET` MUST retrieve the Signal K branch named by the path; e.g.
-`/signalk/v1/api/vessels/123456789/navigation/speedThroughWater` returns
+Making a `GET` request to `/signalk` returns a JSON object which specifies the available Signal K endpoints and some
+information about the server. Also see [versioning](versioning.md) for details about `version` strings.
 
 ```json
 {
-    "value": 2.55,
-    "source": {
-        "type": "NMEA0183",
-        "src": "VHW",
-        "label": "signalk-parser-nmea0183"
+  "endpoints": {
+    "v1": {
+      "version": "1.alpha1",
+      "signalk-http": "http://localhost:3000/signalk/v1/api/",
+      "signalk-ws": "ws://localhost:3000/signalk/v1/stream"
     },
-    "timestamp": "2015-08-31T05:45:36.000Z"
+    "v3": {
+      "version": "3.0",
+      "signalk-http": "http://localhost/signalk/v3/api/",
+      "signalk-ws": "ws://localhost/signalk/v3/stream",
+      "signalk-tcp": "tcp://locahost:8367"
+    }
+  },
+  "server": {
+    "id": "signalk-server-node",
+    "version": "0.1.33"
+  }
 }
 ```
+
+This response is defined by the `discovery.json` schema. In this example, the server supports two versions of the
+specification: `1.alpha1` and `3.0`. For each version, the server indicates which transport protocols it supports and
+the URL that can be used to access that protocol‘s endpoint. Clients should use one of these published endpoints based
+on the protocol version they wish to use.
+
+The server must only return valid URLs and should use IANA standard protocol names such as `http`. However, a server
+may support unofficial protocols and may return additional protocol names; for example, the response above indicates
+the server supports a `signalk-tcp` stream over TCP at on port 8367.
+
+A server may return relative URIs that the client must resolve against the base of the original request.
```

Comment:
Why add this? Not against, more curious. js client does not support this, does it?

Reply:
I didn't add it. This came from http://signalk.org/specification/0.0.1-12/doc/urls_etc.html which was merged in here. The client doesn't support it yet, but it could. I am however OK with removing it.

Reply:
Minor, let's leave it.